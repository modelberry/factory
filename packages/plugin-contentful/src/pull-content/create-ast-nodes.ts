import {
  createContentVarStatement,
  camelToKebab,
  createTsImport,
} from '@modelberry/mbfactory/plain'

export interface CreateAstNodes {
  contentTypeId: string
  varType: string
  varName: string
  contentArray: any[]
}

export const createAstNodes = ({
  contentTypeId,
  varType,
  varName,
  contentArray,
}: CreateAstNodes) => {
  // Add import statements to be added to the main file
  const modelFilenameWithoutExt = `contentful-${camelToKebab(contentTypeId)}`
  const modelImportStatement = createTsImport({
    namedImports: [varType],
    from: `./${modelFilenameWithoutExt}`,
  })
  const contentVarStatement = createContentVarStatement({
    contentArray,
    varName,
    varType,
  })
  return [modelImportStatement, contentVarStatement]
}
