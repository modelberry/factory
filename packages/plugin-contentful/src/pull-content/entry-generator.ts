import { firstUpperCase } from '@modelberry/mbfactory/plain'
import { entriesToJsVariables } from '../lib/entries-to-js-variables'
import { EntriesByContentTypeId } from '../lib/entries-by-content-type-id'

export interface EntryGenerator {
  /** Empty node list, for each generated type an import statement is added */
  entriesByContentTypeId: EntriesByContentTypeId
  localeCode: string
}

export function* entryGenerator({
  entriesByContentTypeId,
  localeCode,
}: EntryGenerator) {
  // Each content type is a file with: export const myType: MyType = [...]
  for (const contentTypeId of Object.keys(entriesByContentTypeId)) {
    const varType = `Contentful${firstUpperCase(contentTypeId)}`
    const varName = contentTypeId
    // Get the content of the js variables from the Contentful entries
    const contentArray: any[] = entriesToJsVariables({
      entryTypeList: entriesByContentTypeId[contentTypeId].entryTypeList,
      localeCode,
    })
    yield { contentTypeId, varType, varName, contentArray }
  }
}
