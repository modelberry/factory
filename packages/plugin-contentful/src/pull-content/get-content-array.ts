import { EntriesByContentTypeId, EntryType } from './entries-by-content-type-id'
import { removeLinkAndLinkType } from './remove-link-and-link-type'

export interface GetContentArray {
  contentTypeId: string
  entriesByContentTypeId: EntriesByContentTypeId
  localeCode: string
}

// Convert the Contentful entries into the a js content array
export const getContentArray = ({
  contentTypeId,
  entriesByContentTypeId,
  localeCode,
}: GetContentArray) => {
  const contentArray: any[] = []
  // Add entries, convert arrays into Collections, etc.
  entriesByContentTypeId[contentTypeId].entryTypeList.forEach(
    (entryType: EntryType) => {
      // Create a new entry with sys.id
      const addEntry: Record<string, any> = {
        sys: { id: entryType.sys.id },
      }
      // Add fields to entry
      Object.keys(entryType.fields).forEach((fieldId: string) => {
        // Remove type and linkType attributes because they can be resolved
        // by lookup up the content type
        const remoteEntryFieldValue = removeLinkAndLinkType({
          fieldValue: entryType.fields[fieldId].fieldValue[localeCode],
        })
        const remoteFieldType = entryType.fields[fieldId].contentFieldType
        if (remoteFieldType?.type === 'Array') {
          const itemsType =
            entryType.fields[fieldId].contentFieldType?.items?.type
          if (itemsType === 'Symbol') {
            // Same as the default below, but more explicit to code it this way
            addEntry[fieldId] = remoteEntryFieldValue
          } else {
            addEntry[`${fieldId}Collection`] = {
              items: remoteEntryFieldValue,
            }
          }
        } else if (remoteFieldType?.type === 'RichText') {
          // TODO: (maybe?) Convert all nodeTypes from string into numbers
          addEntry[fieldId] = {
            json: remoteEntryFieldValue,
          }
        } else {
          addEntry[fieldId] = remoteEntryFieldValue
        }
      })
      contentArray.push(addEntry)
    }
  )
  return contentArray
}
