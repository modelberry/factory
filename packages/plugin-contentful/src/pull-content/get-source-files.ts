import { camelToKebab, createTsExport, Node } from '@modelberry/mbfactory/plain'
import { SourceFile } from '../lib/write-source-files'
import { RemoteEntryIterator } from './remote-entry-generator'
import { createAstNodes } from './create-ast-nodes'

export interface GetSourceFiles {
  remoteEntryIterator: RemoteEntryIterator
  path: string
}

export const getSourceFiles = async ({
  remoteEntryIterator,
  path,
}: GetSourceFiles) => {
  // Generate import statements for each type to be added to the main file
  const allTypesImportStatements: Node[] = []
  const files: SourceFile[] = []
  for await (const entry of remoteEntryIterator) {
    const nodes = createAstNodes(entry)
    const filenameWithoutExt = `contentful-${camelToKebab(
      entry.contentTypeId
    )}-content`
    files.push({
      filename: `${filenameWithoutExt}.ts`,
      nodes,
      path,
    })

    // Add import statements to be added to the main file
    allTypesImportStatements.push(
      createTsExport({
        namedExports: [],
        from: `./${filenameWithoutExt}`,
      })
    )
  }
  files.push({
    filename: 'main-content.ts',
    nodes: allTypesImportStatements,
    path,
  })
  return files
}
