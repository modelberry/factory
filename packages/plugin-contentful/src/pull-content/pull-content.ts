import { Environment } from 'contentful-management/types'
import { Options, Node } from '@modelberry/mbfactory/plain'
import chalk from 'chalk'
import { writeSourceFiles } from '../lib/write-source-files'
import { fetchContentTypes } from '../lib/fetch-content-types'
import { getContentfulLocales } from '../lib/get-contentful-locales'
import { getSourceFiles } from './get-source-files'
import { EntriesByContentTypeId, EntryType } from './entries-by-content-type-id'

export interface PullContent {
  contentfulEnvironment: Environment
  options: Options
  path: string
}

export const pullContent = async ({
  contentfulEnvironment,
  options,
  path,
}: PullContent) => {
  const log = console.log
  const { defaultLocale } = await getContentfulLocales({
    contentfulEnvironment,
  })
  // Use locales in this order, cli overrides all others, remote default
  // locale is a last resort
  const localeCode = options.locale || defaultLocale?.code || 'en-US'
  log(chalk(`- remote default locale: ${defaultLocale?.code}`))
  log(chalk(`- pulling locale: ${localeCode}`))

  const contentTypes = await fetchContentTypes({
    contentfulEnvironment,
    options,
  })
  // Initialize struture to organize entires per content type
  const entriesByContentTypeId: EntriesByContentTypeId = {}
  contentTypes.forEach((contentType) => {
    entriesByContentTypeId[contentType.sys.id] = {
      contentType,
      entryTypeList: [],
    }
  })
  const query = options.type
    ? {
        content_type: 'testTopic',
      }
    : undefined
  // Fetch all entries
  const remoteEntries = await contentfulEnvironment.getEntries(query)
  // Find entries and organize them with the content type
  for (const remoteEntry of remoteEntries.items) {
    const contentTypeId = remoteEntry.sys.contentType.sys.id
    const entryType: EntryType = { sys: remoteEntry.sys, fields: {} }
    // Find fields for content type and entry and organize them together
    for (const fieldId of Object.keys(remoteEntry.fields)) {
      entryType.fields[fieldId] = {
        contentFieldType: entriesByContentTypeId[
          contentTypeId
        ].contentType.fields.find((field: any) => field.id === fieldId),
        fieldValue: remoteEntry.fields[fieldId],
      }
    }
    entriesByContentTypeId[contentTypeId].entryTypeList.push(entryType)
  }
  // Generate import statements for each type to be added to the main file
  const allTypesImportStatements: Node[] = []
  const files = getSourceFiles({
    allTypesImportStatements,
    entriesByContentTypeId,
    localeCode,
    path,
  })
  files.push({
    filename: 'main-content.ts',
    nodes: allTypesImportStatements,
    path,
  })
  await writeSourceFiles({ files, options })
}
