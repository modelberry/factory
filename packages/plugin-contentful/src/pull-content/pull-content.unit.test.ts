jest.mock('fs/promises')

import { writeFile } from 'fs/promises'
import chalk from 'chalk'
import { environmentMock } from '../contentful-mock/contentful-mock'
import { pullContent } from './pull-content'

describe('Pull content should', () => {
  const consoleSpy = jest.spyOn(console, 'log').mockImplementation()
  beforeEach(() => {
    consoleSpy.mockReset()
  })

  test('pull correctly with type filter', async () => {
    await pullContent({
      contentfulEnvironment: environmentMock,
      options: { force: true, type: 'testTopic' },
      path: './dummy',
    })
    expect(writeFile).toMatchSnapshot()
    expect(consoleSpy.mock.calls).toEqual([
      [chalk('- new source file ./dummy/contentful-test-topic.ts')],
    ])
  })
  test('pull correctly without type filter', async () => {
    await pullContent({
      contentfulEnvironment: environmentMock,
      options: { force: true },
      path: './dummy',
    })
    expect(writeFile).toMatchSnapshot()
    expect(consoleSpy.mock.calls).toEqual([
      [chalk('- new source file ./dummy/contentful-test-topic.ts')],
      [chalk('- new source file ./dummy/contentful-test-action.ts')],
      [chalk('- new source file ./dummy/contentful-navigation-section.ts')],
      [chalk('- new source file ./dummy/contentful-topic-section.ts')],
      [chalk('- new source file ./dummy/contentful-topic.ts')],
      [chalk('- new source file ./dummy/contentful-text-section.ts')],
      [chalk('- new source file ./dummy/contentful-navigation-segment.ts')],
      [chalk('- new source file ./dummy/contentful-page.ts')],
      [chalk('- new source file ./dummy/contentful-globals.ts')],
      [chalk('- new source file ./dummy/contentful-embed.ts')],
      [chalk('- new source file ./dummy/contentful-action.ts')],
    ])
  })
})
