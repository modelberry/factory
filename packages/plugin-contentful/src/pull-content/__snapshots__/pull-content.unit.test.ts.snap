// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pull content should pull correctly with type filter 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "./dummy/contentful-test-topic.ts",
      "export const testTopic: ContentfulTestTopic[] = [
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
]
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Pull content should pull correctly without type filter 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "./dummy/contentful-test-topic.ts",
      "export const testTopic: ContentfulTestTopic[] = [
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
]
",
    ],
    Array [
      "./dummy/contentful-test-topic.ts",
      "export const testTopic: ContentfulTestTopic[] = [
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
  {
    abstract: 'Symbol = This is the abstract',
    actions: 'Array = [object Object]',
    heading: 'Symbol = Heres the heading',
  },
]
",
    ],
    Array [
      "./dummy/contentful-test-action.ts",
      "export const testAction: ContentfulTestAction[] = [
  { anchor: 'Symbol = anchor', heading: 'Symbol = Hello' },
]
",
    ],
    Array [
      "./dummy/contentful-navigation-section.ts",
      "export const navigationSection: ContentfulNavigationSection[] = [
  {
    title: 'Symbol = Footer',
    logo: 'Link = [object Object]',
    logoAction: 'Link = [object Object]',
    footer: 'Array = [object Object]',
    social: 'Link = [object Object]',
    actions: 'Link = [object Object]',
    legal: 'Link = [object Object]',
    information: 'Array = [object Object]',
    sitemap:
      'Array = [object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = Header',
    logo: 'Link = [object Object]',
    logoAction: 'Link = [object Object]',
    header: 'Array = [object Object]',
    social: 'Link = [object Object]',
    actions: 'Link = [object Object]',
  },
]
",
    ],
    Array [
      "./dummy/contentful-topic-section.ts",
      "export const topicSection: ContentfulTopicSection[] = [
  {
    title: 'Symbol = Jaco Beach topic section',
    variant: 'Symbol = block',
    topics: 'Array = [object Object],[object Object]',
    eventId: 'Symbol = jaco-beach-topic-section',
  },
  {
    title: 'Symbol = Navigation explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = navigation-explained',
  },
  {
    title: 'Symbol = Text explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = text-explained',
  },
  {
    title: 'Symbol = Card explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = card-explained',
  },
  {
    title: 'Symbol = Divider explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = divider-exmplaed',
  },
  {
    title: 'Symbol = Featured explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = featured-explained',
  },
  {
    title: 'Symbol = Gallery explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = gallery-explained',
  },
  {
    title: 'Symbol = Headline explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = headline-explained',
  },
  {
    title: 'Symbol = Hero explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = hero-explained',
  },
  {
    title: 'Symbol = Video explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = video-explained',
  },
  {
    title: 'Symbol = Image explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = image-explained',
  },
  {
    title: 'Symbol = Quote explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = quote-explained',
  },
  {
    title: 'Symbol = Showcase explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = showcase-explained',
  },
  {
    title: 'Symbol = Block explained',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = block-explained',
  },
  {
    title: 'Symbol = Examples',
    variant: 'Symbol = block',
    topics:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]',
    eventId: 'Symbol = all-variants-exmplained',
  },
  {
    title: 'Symbol = Video examples',
    variant: 'Symbol = video',
    topics: 'Array = [object Object],[object Object]',
    eventId: 'Symbol = video-examples',
  },
  {
    title: 'Symbol = Showcase examples',
    variant: 'Symbol = showcase',
    topics:
      'Array = [object Object],[object Object],[object Object],[object Object]',
    eventId: 'Symbol = showcase-examples',
  },
  {
    title: 'Symbol = Quote examples',
    variant: 'Symbol = quote',
    topics: 'Array = [object Object],[object Object]',
    eventId: 'Symbol = quote-examples',
  },
  {
    title: 'Symbol = Image examples',
    variant: 'Symbol = image',
    topics: 'Array = [object Object],[object Object]',
    eventId: 'Symbol = image-examples',
  },
  {
    title: 'Symbol = Hero examples',
    variant: 'Symbol = hero',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = hero-examples',
  },
  {
    title: 'Symbol = Headline examples',
    variant: 'Symbol = headline',
    topics: 'Array = [object Object]',
    eventId: 'Symbol = headline-examples',
  },
  {
    title: 'Symbol = Gallery examples',
    variant: 'Symbol = gallery',
    topics: 'Array = [object Object],[object Object]',
    eventId: 'Symbol = gallery-examples',
  },
  {
    title: 'Symbol = Featured examples',
    variant: 'Symbol = featured',
    topics: 'Array = [object Object],[object Object]',
    eventId: 'Symbol = featured-examples',
  },
  {
    title: 'Symbol = Divider',
    variant: 'Symbol = divider',
    eventId: 'Symbol = divider-examples',
  },
  {
    title: 'Symbol = Card examples',
    variant: 'Symbol = card',
    topics:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
    eventId: 'Symbol = card-examples',
  },
  {
    title: 'Symbol = Block examples',
    variant: 'Symbol = block',
    topics:
      'Array = [object Object],[object Object],[object Object],[object Object]',
    eventId: 'Symbol = block-examples',
  },
]
",
    ],
    Array [
      "./dummy/contentful-topic.ts",
      "export const topic: ContentfulTopic[] = [
  {
    title: 'Symbol = Jac\\\\u00F3 Beach topic',
    heading: 'Symbol = Jac\\\\u00F3 Beach',
    abstract:
      'Text = Jac\\\\u00F3 is a district of the Garabito canton, in the Puntarenas province of Costa Rica. Jac\\\\u00F3 has a black sand beach that is 4 km (2.5 mi) long and is popular among surfers.',
    media: 'Array = [object Object]',
    icon: 'Symbol = sun',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Another topic',
    heading: 'Symbol = Another topic',
    abstract: 'Text = This is the other topic',
    media: 'Array = [object Object]',
    icon: 'Symbol = archive',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Build great things with Wheelroom',
    heading: 'Symbol = Build great things with Wheelroom',
    abstract:
      'Text = Wheelroom is a new approach to faster, cost efficient and more secure applications.',
    media: 'Array = [object Object]',
    icon: 'Symbol = box',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Image explained',
    heading: 'Symbol = Image',
    abstract: 'Text = Image shows the larger image.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Showcase explained',
    heading: 'Symbol = Showcase',
    abstract:
      \\"Text = Showcase sets your topics on display in a modest way. Stacked on top of each other, it's a clear line-up.\\",
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Headline explained',
    heading: 'Symbol = Headline',
    abstract:
      'Text = Headlines are ideal page openers. Half way on a page they are useful for starting a new section.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Block explained',
    heading: 'Symbol = Block',
    abstract:
      'Text = Block gives overview to a set of topics. Stacked neatly next to each other they provide a clean display.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Card explained',
    heading: 'Symbol = Card',
    abstract: 'Text = Cards are on the table. Like cards are supposed to.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Gallery explained',
    heading: 'Symbol = Gallery',
    abstract: 'Text = Gallery organises images on a grid.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Featured explained',
    heading: 'Symbol = Featured',
    abstract: 'Text = Featured puts it on display. In a spacious room.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Quote explained',
    heading: 'Symbol = Quote',
    abstract: 'Text = Quote. No need to explain that.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Hero explained',
    heading: 'Symbol = Hero',
    abstract: 'Text = Hero puts the message on top of the image.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Divider explained',
    heading: 'Symbol = Divider explained',
    abstract:
      'Text = Diver splits in two. On longer pages the divider provides clean sections.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Video explained',
    heading: 'Symbol = Video',
    abstract:
      'Text = Video moves the image. Upload your own images to the CMS, no need for anything else.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Text explained',
    heading: 'Symbol = Text',
    abstract:
      'Text = Text is free style. Have it your own, including headings, images and all that.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Navigation explained',
    heading: 'Symbol = Navigation',
    abstract:
      'Text = Navigation is for organising links. The best examples are at the top and bottom of this page.',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
  },
  {
    title: 'Symbol = Blue oil',
    heading: 'Symbol = Blue oil',
    abstract:
      'Text = Blue is one of the three primary colours of pigments in painting and traditional colour theory, as well as in the RGB colour model.',
    media:
      'Array = [object Object],[object Object],[object Object],[object Object]',
    icon: 'Symbol = book-open',
    actions: 'Array = [object Object]',
    poster: 'Link = [object Object]',
  },
  {
    title: 'Symbol = Page section examples',
    heading: 'Symbol = Examples',
    abstract:
      'Text = Pages are build up out of sections. Each section has a flavour. We call them display variations.',
    icon: 'Symbol = book-open',
  },
]
",
    ],
    Array [
      "./dummy/contentful-text-section.ts",
      "export const textSection: ContentfulTextSection[] = [
  {
    title: 'Symbol = Test text section',
    variant: 'Symbol = text',
    text: 'RichText = [object Object]',
    eventId: 'Symbol = test-text-section',
  },
]
",
    ],
    Array [
      "./dummy/contentful-navigation-segment.ts",
      "export const navigationSegment: ContentfulNavigationSegment[] = [
  {
    title: 'Symbol = Sitemap segment',
    heading: 'Symbol = Examples',
    abstract: 'Text = All Wheelroom examples',
    actions:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = Actions segment',
    heading: 'Symbol = Actions',
    actions: 'Array = [object Object],[object Object]',
  },
  {
    title: 'Symbol = Menu segment',
    heading: 'Symbol = Wheelroom',
    actions: 'Array = [object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = Social segment',
    heading: 'Symbol = Social',
    actions: 'Array = [object Object]',
  },
]
",
    ],
    Array [
      "./dummy/contentful-page.ts",
      "export const page: ContentfulPage[] = [
  {
    title: 'Symbol = /',
    path: 'Symbol = /',
    sections: 'Array = [object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /another-page',
    path: 'Symbol = /another-page',
    sections: 'Array = [object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples',
    path: 'Symbol = /examples',
    sections: 'Array = [object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/card',
    path: 'Symbol = /examples/card',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/divider',
    path: 'Symbol = /examples/divider',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/featured',
    path: 'Symbol = /examples/featured',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/gallery',
    path: 'Symbol = /examples/gallery',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/headline',
    path: 'Symbol = /examples/headline',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/hero',
    path: 'Symbol = /examples/hero',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/video',
    path: 'Symbol = /examples/video',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/image',
    path: 'Symbol = /examples/image',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/quote',
    path: 'Symbol = /examples/quote',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/showcase',
    path: 'Symbol = /examples/showcase',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/block',
    path: 'Symbol = /examples/block',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/navigation',
    path: 'Symbol = /examples/navigation',
    sections:
      'Array = [object Object],[object Object],[object Object],[object Object],[object Object]',
  },
  {
    title: 'Symbol = /examples/text',
    path: 'Symbol = /examples/text',
    sections: 'Array = [object Object],[object Object],[object Object]',
  },
]
",
    ],
    Array [
      "./dummy/contentful-globals.ts",
      "export const globals: ContentfulGlobals[] = [
  {
    title: 'Symbol = Global settings for this site',
    siteAuthor: 'Symbol = Wheelroom',
    siteDescription:
      'Symbol = Wheelroom is an open source project and created for developers that want to work with the latest technology and services.',
    siteHeading: 'Symbol = Wheelroom',
    siteKeywords: 'Array = Wheelroom,Boilerplate,Example set',
    skipToContentHeading: 'Symbol = Skip to Wheelroom content',
  },
]
",
    ],
    Array [
      "./dummy/contentful-embed.ts",
      "export const embed: ContentfulEmbed[] = []
",
    ],
    Array [
      "./dummy/contentful-action.ts",
      "export const action: ContentfulAction[] = [
  {
    title: 'Symbol = Featured example',
    heading: 'Symbol = Featured example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = featured-example',
  },
  {
    title: 'Symbol = Block example',
    heading: 'Symbol = Block example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = block-example',
  },
  {
    title: 'Symbol = Card example',
    heading: 'Symbol = Card example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = card-example',
  },
  {
    title: 'Symbol = Gallery example',
    heading: 'Symbol = Gallery example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = gallery-example',
  },
  {
    title: 'Symbol = Showcase example',
    heading: 'Symbol = Showcase example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = showcase-example',
  },
  {
    title: 'Symbol = Quote example',
    heading: 'Symbol = Quote example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = quote-example',
  },
  {
    title: 'Symbol = Headline example',
    heading: 'Symbol = Headline example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = headline-example',
  },
  {
    title: 'Symbol = Image example',
    heading: 'Symbol = Image example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = image-example',
  },
  {
    title: 'Symbol = Text example',
    heading: 'Symbol = Text example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = text-example',
  },
  {
    title: 'Symbol = Divider example',
    heading: 'Symbol = Divider example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = divider-example',
  },
  {
    title: 'Symbol = Hero example',
    heading: 'Symbol = Hero example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = hero-example',
  },
  {
    title: 'Symbol = Video example',
    heading: 'Symbol = Video example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = video-example',
  },
  {
    title: 'Symbol = Navigation example',
    heading: 'Symbol = Navigation example',
    page: 'Link = [object Object]',
    eventId: 'Symbol = navigation-example',
  },
  {
    title: 'Symbol = Examples',
    heading: 'Symbol = Examples',
    description:
      'Symbol = An overview of all the Wheelroom page-section exmaples',
    page: 'Link = [object Object]',
    eventId: 'Symbol = check-the-wheelroom-examples-action',
  },
  {
    title: 'Symbol = Home',
    heading: 'Symbol = Home',
    description: 'Symbol = Go to homepage',
    page: 'Link = [object Object]',
    eventId: 'Symbol = home',
  },
  {
    title: 'Symbol = Wheelroom (action)',
    heading: 'Symbol = Wheelroom',
    description: 'Symbol = Go back to the Wheelroom homepage',
    eventId: 'Symbol = wheelroom-action',
  },
  {
    title: 'Symbol = Try Wheelroom for Free (action)',
    heading: 'Symbol = Try Wheelroom for Free',
    description: 'Symbol = Visit our Github Wheelroom code repository',
    url: 'Symbol = https://github.com/wheelroom/wheelroom',
  },
  {
    title: 'Symbol = Github',
    heading: 'Symbol = Github',
    description: 'Symbol = Visit our Github Wheelroom code repository',
    icon: 'Symbol = Github',
    url: 'Symbol = https://github.com/wheelroom/wheelroom',
    eventId: 'Symbol = github',
  },
  {
    title: 'Symbol = Another action',
    heading: 'Symbol = Another action',
    page: 'Link = [object Object]',
    eventId: 'Symbol = another-action',
  },
  {
    title: 'Symbol = Jac\\\\u00F3 Beach on Wikipedia',
    heading: 'Symbol = Jac\\\\u00F3 Beach on Wikipedia',
    description:
      'Symbol = Jac\\\\u00F3 was created on 16 September 1965. Segregated from Puntarenas.',
    icon: 'Symbol = external-link',
    url: 'Symbol = https://en.wikipedia.org/wiki/Jac\\\\u00F3,_Costa_Rica',
    eventId: 'Symbol = jaco-beach',
  },
]
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
